Fix issue #31

rodrigopavezi opened the pull request:
@tejohnso please review. cheers

rodrigopavezi committed:
Added a check for ensuring firebase app has not been initialized

rodrigopavezi committed:
Fixed tests

rodrigopavezi committed:
Fixed linting and built

tejohnso commented on rise-financial-es6.js:
```diff
@@ -294,7 +294,8 @@
 
-      if ( !this._firebaseApp ) {
+      // ensure firebase app has not been initialized already
+      if ( !this._firebaseApp && !firebase.apps.length ) {
```
@rodrigopavezi If firebase has allready been initialized and the second test fails then this._firebaseApp isn't set. Shouldn't it be set to the existing instance that caused the second test to fail?

rodrigopavezi committed:
Fixed firebase app initialization

Changed JSONP callback to be unique between components instances
Updated tests

rodrigopavezi commented:
@tejohnso good catch. I have changed that. I have also fixed another issue with the JSONP callback so it will be unique per instance of the component. Please review. cheers

rodrigopavezi committed:
Changed the unique value to be the component instance ID

stulees reviewed (changes requested):
@rodrigopavezi This needs a version bump and rebuild, thanks.

rodrigopavezi committed:
Version Bumped

rodrigopavezi commented:
@stulees version bumped, please review. cheers

stulees commented:
@rodrigopavezi 👍
